src/Dijkstra/AST/Examples/Maybe/Branching.agda

```diff
@@ -16,33 +16,196 @@ module Dijkstra.AST.Examples.Maybe.Branching where
+
+      ASTPredTrans.opPT BranchPT (Right (BCif c)) f P i
+
+    requires two proofs, one for when the condition c is true and one for when it's false; each case
+    gets evidence that the condition evaluates to the relevant boolean.
+
+    Furthermore, we can see that the goals for each conjunct are now derived from the postcondition,
+    bpPost.  Now we can start to see what we need to prove.
+
+    The second conjunct does not depend on the first, so we can separate into two goals:
+
+      proj₁ (branchingProgWP _) = ?
+      proj₂ (branchingProgWP _) = ?
+
+    Using C-c C-, in the first hole gives us this goal:
+
+      ⌊ n ≟ℕ 0 ⌋ ≡ true → n ≡ 0
+
+    This is easily dispatched with toWitnessT.
+
+      proj₁ (branchingProgWP _) = toWitnessT
+
+    The second conjunct is similar.  Doing C-c C-, in the hole shows:
+
+      ⌊ n ≟ℕ 0 ⌋ ≡ false → 0 < n × n + (n + zero) ≡ n + (n + zero)
+
+    Let's give the evidence that the condition is false a name, so we can
+    prove the two conjuncts separately:
+
+      proj₂ (branchingProgWP _) isFalse = ?
+
+    The goal shown by C-c C-, is now:
+
+      0 < n × n + (n + zero) ≡ n + (n + zero)
+
+    We can do C-c C-r to refine this to two separate goals, the first of which is 0 < n.  Again, we
+    use the relevant toWitness function (toWitnessF because the condition is false), along with a
+    library function n≢0⇒n>0 to fill the hole for the first conjunct.
+
+      proj₂ (branchingProgWP _) isFalse = (n≢0⇒n>0 (toWitnessF isFalse)) , ?
+
+    C-c C-, shows that the second hole needs evidence that n + (n + zero) ≡ n + (n + zero), which is
+    easily satisfied by refl, resulting in the complete proof below.
+
+    One important lesson is to remember to use the variants of C-c C-, with zero, one or two C-u's
+    before it.  Each is the most useful of the three in some contexts.  Experienced users might
+    predict which one is best, but just having enough experience to remember to try all three is a
+    big help.
+
+    -}
+
+    proj₁ (branchingProgWP _) isTrue  =           toWitnessT isTrue
+    proj₂ (branchingProgWP _) isFalse = (n≢0⇒n>0 (toWitnessF isFalse)) , refl
+
+    -- And therefore, the result of running the program satisfies the postcondition
+    prop : (i : Input) → bpPost (runMaybeExt branchingProg i)
+    prop i =
+      ASTSufficientPT.sufficient MaybeSufExt branchingProg bpPost i (branchingProgWP i)
+
+  module Prettier where
+    open BranchingSyntax MaybeOps
+    open MaybeBranchingSyntax
+
+    -- Same program with nicer syntax using ifAST, bail and return
+    branchingProg : MaybeDExt ℕ
+    branchingProg = ifAST ⌊ n ≟ℕ 0 ⌋
+                    then bail
+                    else (return (2 * n))
+
+    --  Note that the same proof works for both versions (as they are equivalent)
+    branchingProgWP : (i : Input)
+                         → ASTPredTrans.predTrans MaybePTExt branchingProg bpPost i
+    proj₁ (branchingProgWP i) isTrue  =           toWitnessT isTrue
```

Comment:
replace `i` with `_` here and the next line (to be consistent with "non" pretty one

Reply:
DONE